### PR TITLE
ti_misp: harmonise object fields between datastreams

### LINKED ISSUE #5834: [misp] Field conflicts between threat and threat_attributes
There were 2 new field conflicts introduced after the threat_attributes integration was released and used.
image

misp.event.distribution
image

misp.object.timestamp
image

The misp.event.distribution will be an easy change in the fields.yml for that specific data stream (I recommend keyword as that is what the original MISP integration has and we don't see the value of this being a long as it usually only contains the number 1 ), however, the misp.object.timestamp might be a little more tricky as neither of the integrations seemed to have mapped that field.

@P1llus and @legoguy1000 , could you guys please address this?

Here is an example event with the misp.object.timestamp:
```
{"Event":{"Attribute":[],"CryptographicKey":[],"EventReport":[],"Galaxy":[],"Object":{"Attribute":{"Galaxy":[],"ShadowAttribute":[],"Tag":[{"colour":"#27c1f2","exportable":true,"hide_tag":false,"id":"49","is_custom_galaxy":false,"is_galaxy":false,"local":0,"local_only":false,"name":"Diamond: Capability/TTP","numerical_value":null,"relationship_type":null,"user_id":"0"},{"colour":"#b7bfe8","exportable":true,"hide_tag":false,"id":"16","is_custom_galaxy":false,"is_galaxy":false,"local":0,"local_only":false,"name":"killchain:Delivery","numerical_value":null,"relationship_type":null,"user_id":"0"}],"category":"Network activity","comment":"","deleted":false,"disable_correlation":false,"distribution":"5","event_id":"12874","first_seen":null,"id":"307473","last_seen":null,"object_id":"18647","object_relation":"url","sharing_group_id":"0","timestamp":"1675782548","to_ids":true,"type":"url","uuid":"ebe85f63-1ff2-482b-a1bb-2dcd34f4bebb","value":"/wp\\-includes/js/jquery.min.js"},"ObjectReference":[],"comment":"","deleted":false,"description":"url object describes an url along with its normalized field (like extracted using faup parsing library) and its metadata.","distribution":"5","event_id":"12874","first_seen":null,"id":"18647","last_seen":null,"meta-category":"network","name":"url","sharing_group_id":"0","template_uuid":"60efb77b-40b5-4c46-871b-ed1ed999fce5","template_version":"9","timestamp":"1675782559","uuid":"d950dfa2-9f20-4aa7-b347-6e7d313cdb69"},"Org":{"id":"4","local":true,"name":"Organization SOC","uuid":"9158b048-9b84-4a7e-9aec-57120f3ea268"},"Orgc":{"id":"4","local":true,"name":"Organization SOC","uuid":"9158b048-9b84-4a7e-9aec-57120f3ea268"},"RelatedEvent":[],"ShadowAttribute":[],"Tag":[{"colour":"#33FF00","exportable":true,"hide_tag":false,"id":"13","is_custom_galaxy":false,"is_galaxy":false,"local":0,"local_only":false,"name":"tlp:green","numerical_value":null,"relationship_type":null,"user_id":"0"},{"colour":"#f29035","exportable":true,"hide_tag":false,"id":"65","is_custom_galaxy":false,"is_galaxy":false,"local":0,"local_only":false,"name":"veris:action:social","numerical_value":null,"relationship_type":null,"user_id":"0"},{"colour":"#17a4d1","exportable":true,"hide_tag":false,"id":"62","is_custom_galaxy":false,"is_galaxy":false,"local":0,"local_only":false,"name":"veris:actor:external","numerical_value":null,"relationship_type":null,"user_id":"0"}],"analysis":"0","attribute_count":"3","date":"2023-02-03","disable_correlation":false,"distribution":"1","extends_uuid":"","id":"12874","info":"Phishing - Payment Processed for Inovice75 on Feb 01,2023","locked":false,"org_id":"4","orgc_id":"4","proposal_email_lock":false,"protected":null,"publish_timestamp":"0","published":false,"sharing_group_id":"0","threat_level_id":"3","timestamp":"1675782577","uuid":"0d6aba79-dd30-4ffb-b90e-81a8322517d0"}}
```

### PR DESCRIPTION
**Pull request by efd6**
## What does this PR do?

Brings field definitions into agreement. This is the second part of addressing #5834,

Note that if the event added in tests here is add to the threat datastream test, it fails due to a failure of the uri_parts processor to handle `\` in the host part of a URL.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ]

## Related issues

- Fixes #5834.

### CODE REVIEW

File: packages/ti_misp/data_stream/threat_attributes/_dev/test/pipeline/test-misp-sample-ndjson.log [hunk 1]
```diff
@@ -10,3 +10,4 @@
...
...4-4a7e-9aec-57120f3ea268"},"RelatedEvent":[],"ShadowAttribute":[],"Tag":[{"colour":"#33FF00","exportable":true,"hide_tag":false,"id":"13","is_custom_galaxy":false,"is_galaxy":false,"local":0,"local_only":false,"name":"tlp:green","numerical_value":null,"relationship_type":null,"user_id":"0"},{"colour":"#f29035","exportable":true,"hide_tag":false,"id":"65","is_custom_galaxy":false,"is_galaxy":false,"local":0,"local_only":false,"name":"veris:action:social","numerical_value":null,"relationship_type":null,"user_id":"0"},{"colour":"#17a4d1","exportable":true,"hide_tag":false,"id":"62","is_custom_galaxy":false,"is_galaxy":false,"local":0,"local_only":false,"name":"veris:actor:external","numerical_value":null,"relationship_type":null,"user_id":"0"}],"analysis":"0","attribute_count":"3","date":"2023-02-03","disable_correlation":false,"distribution":"1","extends_uuid":"","id":"12874","info":"Phishing - Payment Processed for Inovice75 on Feb 01,2023","locked":false,"org_id":"4","orgc_id":"4","proposal_email_lock":false,"protected":null,"publish_timestamp":"0","published":false,"sharing_group_id":"0","threat_level_id":"3","timestamp":"1675782577","uuid":"0d6aba79-dd30-4ffb-b90e-81a8322517d0"}}
```
Comment (kcreddy) [inline packages/ti_misp/data_stream/threat_attributes/_dev/test/pipeline/test-misp-sample-ndjson.log:13]: This response doesn't seem to belong to [threat attributes API](https://www.misp-project.org/openapi/#tag/Attributes/operation/restSearchAttributes)

Was this supposed to be added to `threat` datastream instead?

File: packages/ti_misp/data_stream/threat_attributes/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -114,6 +114,55 @@ processors:
...
+      ignore_missing: true
+      if: 'ctx.misp?.event?.Attribute != null && ctx.misp.event.Attribute.size() == 0'
+  - remove:
+      field:
+        - misp.event.Object
+      ignore_missing: true
+      if: 'ctx.misp?.event?.Object != null && ctx.misp.event.Object.size() == 0'
+  - rename:
+      field: misp.event.Attribute
+      target_field: misp.attribute
```
Comment (kcreddy) [inline packages/ti_misp/data_stream/threat_attributes/elasticsearch/ingest_pipeline/default.yml:143]: We are already renaming `event.original` to `misp.attribute` earlier in the pipeline.

**Review by kcreddy — COMMENTED** [review]
Need clarification about sample log

File: packages/ti_misp/data_stream/threat_attributes/_dev/test/pipeline/test-misp-sample-ndjson.log (hunk 1 above)
Comment (efd6) [reply packages/ti_misp/data_stream/threat_attributes/_dev/test/pipeline/test-misp-sample-ndjson.log:13]: Yes, good catch. Thanks

File: packages/ti_misp/data_stream/threat_attributes/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (efd6) [reply packages/ti_misp/data_stream/threat_attributes/elasticsearch/ingest_pipeline/default.yml]: Fixed.

Comment (kcreddy) [reply packages/ti_misp/data_stream/threat_attributes/elasticsearch/ingest_pipeline/default.yml]: Sorry, what I actually meant to say was that all these renaming with `misp.event.*` fields never exists in the Attributes API. It has fields named `misp.attribute.*` i.e., `misp.attribute.Object`, `misp.attribute.Event`, `misp.attribute.Tag` which were already renamed in the pipeline earlier.

Comment (efd6) [reply packages/ti_misp/data_stream/threat_attributes/elasticsearch/ingest_pipeline/default.yml]: Indeed, you're correct, fixed fixed.

File: packages/ti_misp/changelog.yml [hunk 3]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.13.1"
+  changes:
+    - description: Harmonise object and org fields.
```
Comment (kcreddy) [inline packages/ti_misp/changelog.yml:4]: `org` is probably not relevant anymore in this PR

**Review by kcreddy — APPROVED** [review]
LGTM 👍🏼 Minor suggestion

File: packages/ti_misp/changelog.yml (hunk 3 above)
Comment (efd6) [inline packages/ti_misp/changelog.yml:4]:
```suggestion
    - description: Harmonise object fields in data streams.
```

**Pull request merged**